lib/lambda.js

```diff
@@ -18,17 +282,20 @@ function elasticApmAwsLambda (agent) {
         logStreamName: context.logStreamName,
         executionEnv: process.env.AWS_EXECUTION_ENV,
         region: process.env.AWS_REGION,
-        input: payload,
+        input: event,
         output: result
       }
     })
   }
 
-  function wrapContext (trans, payload, context) {
+  function wrapContext (trans, event, context) {
     shimmer.wrap(context, 'succeed', (succeed) => {
       return function wrappedSucceed (result) {
+        if (result && result.statusCode && result.statusCode >= 500 && isApiGatewayEvent(event)) {
+          trans.result = 'HTTP ' + result.statusCode.toString()[0] + 'xx'
+        }
```

Comment:
Yes, we look for a case-insensitive `statuscode` key in the response. If present, we convert it to an `HTTP Xxx` and set that in `transaction.result`. So, if the function decides it's a `5xx` then we'll set `5xx` even without an exception. [code](https://github.com/elastic/apm-agent-python/blob/e989439a2702e1cf8256687e6ba1caf8dc20b44d/elasticapm/contrib/serverless/aws.py#L166-L176)

Reply:
Thanks for the note on the case insensitivity @basepi, I'd missed that.  Code updated in 0e250479e5bd31e7cecec11075e696afea5c6fe5

Reply:
@astorm I think this `trans.result = ...` handling will also need to be in `wrapLambdaCallback`. IIUC, when a callback is used the Lambda runtime does *not* call our wrapped `context.succeed()`.

Reply:
Sure thing trent, done in 236f6193504ef6da0cbb7d3bb4acb440dadbb41a

Reply:
@basepi I'm curious why the case-insensitive check. My read of https://docs.aws.amazon.com/apigateway/latest/developerguide/handle-errors-in-lambda-integration.html is that a Lambda "proxy" integration with API Gateway (which, IIUC, is what we are talking about; vs. "custom" integrations) must have a result object with this format:

```
{
  "isBase64Encoded" : "boolean",
  "statusCode": "number",
  "headers": { ... },
  "body": "JSON string"
}
```

Actually, this looks like a better reference: https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-output-format

Reply:
@trentm I don't actually have a good answer. I guess I was trying to be flexible, because `statusCode` seemed like a convention more than a requirement. That said, I just tested with `statuscode` and it threw an internal server error. So I'm going to tear out my case-insensitivity. @astorm I recommend you do the same, sorry for misguiding you with my weird assumptions. :flushed:

Reply:
Okay, that makes sense. Thanks very much for checking.